oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/common/dao/JDBCContinuousProfilingPolicyDAO.java

```diff
@@ -0,0 +1,99 @@
+import org.apache.skywalking.oap.server.core.storage.type.HashMapConverter;
+import org.apache.skywalking.oap.server.library.client.jdbc.hikaricp.JDBCClient;
+import org.apache.skywalking.oap.server.library.util.CollectionUtils;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.SQLExecutor;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.TableMetaInfo;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.common.JDBCTableInstaller;
+import org.apache.skywalking.oap.server.storage.plugin.jdbc.common.TableHelper;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class JDBCContinuousProfilingPolicyDAO extends JDBCSQLExecutor implements IContinuousProfilingPolicyDAO {
+    private final JDBCClient jdbcClient;
+    private final TableHelper tableHelper;
+
+    @Override
+    public void savePolicy(ContinuousProfilingPolicy policy) throws IOException {
+        final List<ContinuousProfilingPolicy> existingPolicy = queryPolicies(Arrays.asList(policy.getServiceId()));
+        SQLExecutor sqlExecutor;
+        final var model = TableMetaInfo.get(ContinuousProfilingPolicy.INDEX_NAME);
+        if (CollectionUtils.isNotEmpty(existingPolicy)) {
+            sqlExecutor = getUpdateExecutor(model, policy, 0, new ContinuousProfilingPolicy.Builder(), null);
+        } else {
+            sqlExecutor = getInsertExecutor(model, policy, 0,
+                new ContinuousProfilingPolicy.Builder(), new HashMapConverter.ToStorage(), null);
+        }
+
+        try (Connection connection = jdbcClient.getConnection()) {
+            sqlExecutor.invoke(connection);
+        } catch (SQLException e) {
+            throw new IOException(e);
+        }
+    }
+
+    @SneakyThrows
+    @Override
+    public List<ContinuousProfilingPolicy> queryPolicies(List<String> serviceIdList) throws IOException {
+        final var tables = tableHelper.getTablesWithinTTL(ContinuousProfilingPolicy.INDEX_NAME);
+        List<Object> condition = new ArrayList<>();
+        condition.add(ContinuousProfilingPolicy.INDEX_NAME);
+        condition.addAll(serviceIdList);
+
+        StringBuilder whereQuery = new StringBuilder()
+            .append(" where ")
+            .append(JDBCTableInstaller.TABLE_COLUMN).append(" = ?")
+            .append(" and ").append(ContinuousProfilingPolicy.SERVICE_ID)
+            .append(" in ").append(serviceIdList.stream().map(s -> "?").collect(Collectors.joining(",", "(", ")")));
+        for (String table : tables) {
+            return jdbcClient.executeQuery("select * from " + table + whereQuery, this::buildPolicies, condition.toArray(new Object[0]));
+        }
+
+        return Collections.emptyList();
```

Comment:
This is not fixed. You might want to change to 

```suggestion
        final var results = new ArrayList<ContinuousProfilingPolicy>();
        for (String table : tables) {
            results.addAll(jdbcClient.executeQuery("select * from " + table + whereQuery, this::buildPolicies, condition.toArray(new Object[0])));
        }
        return results;
```

Reply:
Ohh, I forget to change. been updated, please re-check. :)